vernac/obligations.ml

```diff
@@ -20,6 +20,12 @@ open Pp
 open CErrors
 open Util
 
+(* EJGA: This should go away, no more need for fix_exn  *)
+module Stm = struct
+    let u = (fun x -> x)
+    let get_fix_exn () = u
+end
+
```

Comment:
This is wrong, get_fix_exn cannot be dummy.  It is a hack to fix a bug: 77cf18eb844b45776b2ec67be9f71e8dd4ca002c
As you can see in the commit, `~fix_exn` is passed twice.  In one case it is dummy, in the other one it is not.

Please put the `fix_exn` hack back in the API of the `Stm`.

The `fix_exn` thing is important for the following reason: OCaml is strict.
If you have code like `try x := f a with e -> raise (fix e)` and you turn it into `try x := lazy (f y) with e -> raise (fix e)` you change the semantics: the exception `e` is only fixed in the first case.  In my case `fix` is attaching to `e` the `state_id` to which it belongs.  When the `Stm` creates a `Future.t` it always uses the `enx_on` function as `fix_exn` to ensure the eventual exceptions are localized to the right state id.  IIRC, the `Obligations` plugin does mix&match futures, and it needs the "current" function to `fix_exn`.  The horrible API in `Stm` is to let the `Obligations` code fetch it.